crypto/src/proofs/transparent.rs

```diff
@@ -422,15 +423,18 @@ impl SwapProof {
     pub fn verify(
         &self,
         anchor: tct::builder::block::Root,
-        value_commitment: value::Commitment,
+        swap_nft_value_commitment: value::Commitment,
         nullifier: Nullifier,
-        rk: VerificationKey<SpendAuth>,
     ) -> anyhow::Result<(), Error> {
-        // Note commitment integrity.
+        // Swap NFT note commitment integrity.
         let s_component_transmission_key = Fq::from_bytes(self.pk_d.0);
         if let Ok(transmission_key_s) = s_component_transmission_key {
-            let note_commitment_test =
-                note::commitment(self.note_blinding, self.value, self.g_d, transmission_key_s);
+            let note_commitment_test = note::commitment(
+                self.note_blinding,
+                self.swap_nft_value,
```

Comment:
Here this should hardcode `1` as the amount, and just include the asset ID in the witness data: https://github.com/penumbra-zone/penumbra/pull/1104/files#r925088722